vllm/model_executor/input_metadata.py

```diff
@@ -1,91 +1,42 @@
-from typing import Dict, List, Optional, Tuple
+from typing import List, Optional
 
 import torch
-from xformers.ops import AttentionBias
-
-from vllm.sampling_params import SamplingParams, SamplingType
-from vllm.sequence import SequenceData
 
 
 class InputMetadata:
-    """Metadata for input sequences. Used for PagedAttention.
+    """Metadata for input sequences. Used in PagedAttention.
 
     Args:
-        seq_groups: List of (seq_ids, sampling_params).
-        seq_data: Seq_id -> SequenceData.
         prompt_lens: Lengths of prompts.
         slot_mapping: The address to write the new KV to of each token.
-        context_lens: the length of attention context for each generation token.
         max_context_len: The maximum context length.
+        context_lens: the length of attention context for each sequence.
         block_tables: The block tables. (Seq id -> list of physical block)
     """
 
     def __init__(
         self,
-        seq_groups: List[Tuple[List[int], SamplingParams]],
-        seq_data: Dict[int, SequenceData],
         prompt_lens: List[int],
         slot_mapping: torch.Tensor,
```

Comment:
maybe not need to resolved in this PR, but can we move `positions`, `kv_caches`, and `cache_events` into `input_metadata`? In this way adding a new model can be slight cleaner.

Reply:
Good point. 
1. I believe `positions` should be a regular input rather than input metadata, as it is standard input to LLMs (which is also used in training).
2.  For `kv_caches`, I think either makes sense.
3. `cache_events` will be deleted in the next PR.

If you don't have strong preference on `kv_caches`, let's not change it in this PR.

Reply:
Sounds good!